clang/Cursor.d

```diff
@@ -105,7 +169,184 @@ struct Cursor
+            "__DBL_MAX_EXP__" : 0,
+            "__DBL_MAX__" : 0,
+            "__DBL_MIN_10_EXP__" : 0,
+            "__DBL_MIN_EXP__" : 0,
+            "__DBL_MIN__" : 0,
+            "__LDBL_DENORM_MIN__" : 0,
+            "__LDBL_HAS_DENORM__" : 0,
+            "__LDBL_DIG__" : 0,
+            "__LDBL_EPSILON__" : 0,
+            "__LDBL_HAS_INFINITY__" : 0,
+            "__LDBL_HAS_QUIET_NAN__" : 0,
+            "__LDBL_MANT_DIG__" : 0,
+            "__LDBL_MAX_10_EXP__" : 0,
+            "__LDBL_MAX_EXP__" : 0,
+            "__LDBL_MAX__" : 0,
+            "__LDBL_MIN_10_EXP__" : 0,
+            "__LDBL_MIN_EXP__" : 0,
+            "__LDBL_MIN__" : 0,
+            "__POINTER_WIDTH__" : 0,
+            "__WINT_UNSIGNED__" : 0,
+            "__INT8_TYPE__" : 0,
+            "__INT16_TYPE__" : 0,
+            "__INT32_TYPE__" : 0,
+            "__INT64_TYPE__" : 0,
+            "__INT64_C_SUFFIX__" : 0,
+            "__USER_LABEL_PREFIX__" : 0,
+            "__FINITE_MATH_ONLY__" : 0,
+            "__GNUC_STDC_INLINE__" : 0,
+            "__GCC_ATOMIC_TEST_AND_SET_TRUEVAL" : 0,
+            "__GCC_ATOMIC_BOOL_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_CHAR_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_CHAR16_T_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_CHAR32_T_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_WCHAR_T_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_SHORT_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_INT_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_LONG_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_LLONG_LOCK_FREE" : 0,
+            "__GCC_ATOMIC_POINTER_LOCK_FREE" : 0,
+            "__NO_INLINE__" : 0,
+            "__FLT_EVAL_METHOD__" : 0,
+            "__FLT_RADIX__" : 0,
+            "__DECIMAL_DIG__" : 0,
+            "__amd64__" : 0,
+            "__amd64" : 0,
+            "__x86_64" : 0,
+            "__x86_64__" : 0,
+            "__k8" : 0,
+            "__k8__" : 0,
+            "__tune_k8__" : 0,
+            "__LITTLE_ENDIAN__" : 0,
+            "__REGISTER_PREFIX__" : 0,
+            "__NO_MATH_INLINES" : 0,
+            "__SSE2__" : 0,
+            "__SSE2_MATH__" : 0,
+            "__SSE__" : 0,
+            "__SSE_MATH__" : 0,
+            "__MMX__" : 0,
+            "__GCC_HAVE_SYNC_COMPARE_AND_SWAP_1" : 0,
+            "__GCC_HAVE_SYNC_COMPARE_AND_SWAP_2" : 0,
+            "__GCC_HAVE_SYNC_COMPARE_AND_SWAP_4" : 0,
+            "__GCC_HAVE_SYNC_COMPARE_AND_SWAP_8" : 0,
+            "__unix" : 0,
+            "__unix__" : 0,
+            "__linux" : 0,
+            "__linux__" : 0,
+            "__gnu_linux__" : 0,
+            "__ELF__" : 0,
+            "__STDC__" : 0,
+            "__STDC_HOSTED__" : 0,
+            "__STDC_VERSION__" : 0,
+            "__STDC_UTF_16__" : 0,
+            "__STDC_UTF_32__" : 0,
+            "__int128_t" : 0,
+            "__uint128_t" : 0,
+            "__builtin_va_list" : 0,
+            "unix" : 0,
+            "linux" : 0,
+        ];
+
+        return (spelling in predefined) != null;
```

Comment:
`null` should always be compared with `is`, or let the implicit cast to bool handle it.